src/i18n.ts

```diff
@@ -201,10 +201,10 @@ should either enable sessions or use `ctx.i18n.useLocale()` instead.",
  *
  * @param key Key of the message to listen for.
  */
-export function hears(key: string) {
-  return function <C extends Context & I18nFlavor>(
-    ctx: C,
-  ): ctx is HearsContext<C> {
+export function hears<C extends Context>(key: string) {
+  return function <TContext extends I18nFlavor<C>>(
```

Comment:
I wonder if this is better than removing the type parameter from `hears` and then doing
```suggestion
  return function <C extends Context, CTContext extends I18nFlavor<C>>(
```